replica/replica.go

```diff
@@ -1413,3 +1414,12 @@ func (r *Replica) SetReplicaMode(mode string) error {
 	}
 	return nil
 }
+
+// SetCheckpoint ...
+func (r *Replica) SetCheckpoint(snapshotName string) error {
+	r.Lock()
+	defer r.Unlock()
+	r.info.Checkpoint = snapshotName
+
+	return r.encodeToFile(&r.info, volumeMetaData)
```

Comment:
hope we have a testcase for upgrade scenario

Reply:
Upgrade test is already present, still adding one more test to upgrade from 1.11 release